x-pack/test/detection_engine_api_integration/utils/create_prebuilt_rule_saved_objects.ts

```diff
@@ -10,61 +10,80 @@ import {
-      tags: ['test-tag-1'],
-      enabled: true,
-    },
-    type: 'security-rule',
-    references: [],
-    coreMigrationVersion: '8.6.0',
-    updated_at: '2022-11-01T12:56:39.717Z',
-    created_at: '2022-11-01T12:56:39.717Z',
-  },
-  {
-    'security-rule': {
-      ...getPrebuiltRuleMock(),
-      rule_id: '000047bb-b27a-47ec-8b62-ef1a5d2c9e19',
-      tags: ['test-tag-2'],
-    },
-    type: 'security-rule',
-    references: [],
-    coreMigrationVersion: '8.6.0',
-    updated_at: '2022-11-01T12:56:39.717Z',
-    created_at: '2022-11-01T12:56:39.717Z',
-  },
-  {
-    'security-rule': {
-      ...getPrebuiltRuleMock(),
-      rule_id: '00140285-b827-4aee-aa09-8113f58a08f3',
-      tags: ['test-tag-3'],
-    },
-    type: 'security-rule',
-    references: [],
-    coreMigrationVersion: '8.6.0',
-    updated_at: '2022-11-01T12:56:39.717Z',
-    created_at: '2022-11-01T12:56:39.717Z',
-  },
+  createRuleSavedObject({
+    ...getPrebuiltRuleWithExceptionsMock(),
+    rule_id: ELASTIC_SECURITY_RULE_ID,
+    tags: ['test-tag-1'],
+    enabled: true,
+  }),
+  createRuleSavedObject({
+    ...getPrebuiltRuleMock(),
+    rule_id: '000047bb-b27a-47ec-8b62-ef1a5d2c9e19',
+    tags: ['test-tag-2'],
+  }),
+  createRuleSavedObject({
+    ...getPrebuiltRuleMock(),
+    rule_id: '00140285-b827-4aee-aa09-8113f58a08f3',
+    tags: ['test-tag-3'],
+  }),
+];
+
+export const SAMPLE_PREBUILT_RULES_WITH_HISTORICAL_VERSIONS = [
+  createRuleSavedObject({
+    ...getPrebuiltRuleMock(),
+    rule_id: ELASTIC_SECURITY_RULE_ID,
+    version: 1,
+  }),
+  createRuleSavedObject({
+    ...getPrebuiltRuleMock(),
+    rule_id: ELASTIC_SECURITY_RULE_ID,
+    version: 2,
+  }),
+  createRuleSavedObject({ ...getPrebuiltRuleMock(), rule_id: 'rule-2', version: 1 }),
+  createRuleSavedObject({ ...getPrebuiltRuleMock(), rule_id: 'rule-2', version: 2 }),
+  createRuleSavedObject({ ...getPrebuiltRuleMock(), rule_id: 'rule-2', version: 3 }),
 ];
 
+export const UNIQUE_HISTORICAL_RULES_COUNT = new Set(
+  SAMPLE_PREBUILT_RULES_WITH_HISTORICAL_VERSIONS.map((rule) => rule['security-rule'].rule_id)
+).size;
+
+export type RulesPackageType = 'historical' | 'latest_only';
+
 /**
  * Creates saved objects with prebuilt rule assets which can be used for installing actual prebuilt rules after that.
  *
  * @param es Elasticsearch client
  */
-export const createPrebuiltRuleAssetSavedObjects = async (es: Client): Promise<void> => {
+export const createPrebuiltRuleAssetSavedObjects = async (
+  es: Client,
+  packageType: RulesPackageType = 'latest_only'
+): Promise<void> => {
+  const rulesToCreate =
+    packageType === 'historical'
+      ? SAMPLE_PREBUILT_RULES_WITH_HISTORICAL_VERSIONS
+      : SAMPLE_PREBUILT_RULES;
+
   await es.bulk({
-    refresh: 'wait_for',
-    body: SAMPLE_PREBUILT_RULES.flatMap((doc) => [
+    refresh: true,
+    body: rulesToCreate.flatMap((doc) => [
       { index: { _index: '.kibana', _id: `security-rule:${doc['security-rule'].rule_id}` } },
```

Comment:
In the case of historical rules the `_id` must include the version. I guess the newly added tests for historical rules still work, but no actual historical versions are created -- only the "latest" ones for each rule. This could be a reason for adding an explicit check for the number of assets created after calling `es.bulk`.

Reply:
Nice catch 👍 I added a separate method for historical rule creation.